pkg/admission/clusterworkspaceshard/admission.go

```diff
@@ -125,6 +138,10 @@ func (o *clusterWorkspaceShard) Admit(ctx context.Context, a admission.Attribute
 	return nil
 }
 
+func (o *clusterWorkspaceShard) SetShardBaseURLProvider(shardBaseURLProvider func() string) {
+	o.shardBaseURLProvider = shardBaseURLProvider
```

Comment:
I don't think you need this provider hack here. We only needed it because the external address is not known early enough when admission is setup. Just a simple url.URL or string should do.

Reply:
Updated